app/util/cron.py

```diff
@@ -0,0 +1,58 @@
+import atexit
+import datetime
+import json
+import logging
+import sys
+
+from app import app
+from sqlalchemy import func
+from app.model.email import Email, EmailStatus
+from app.util.db import db
+from app.util.smtp import SMTP
+from apscheduler.schedulers.background import BackgroundScheduler
+from apscheduler.triggers.interval import IntervalTrigger
+
+logging.basicConfig()
+SEND_INTERVAL_SEC = 60
+
+
+def resend_mail():
+    with app.app_context():
+        since = datetime.datetime.utcnow() - datetime.timedelta(hours=1)
+        mails_failed = EmailStatus.query \
+            .filter(EmailStatus.delivered == False, EmailStatus.sent_at > since) \
+            .having(func.count_(EmailStatus.delivered) <= 2) \
+            .group_by(EmailStatus.email_id, EmailStatus.delivered).all()
+        result = EmailStatus.query.filter(EmailStatus.delivered == True, EmailStatus.sent_at > since).all()
```

Comment:
can't we fetch just the failed messages? if a message failed to be delivered more than X times in the last Y hours how can it turn up in mails_success?

Reply:
i rewrote algorithm for getting non-delivered messages. please check it out now